### PR TITLE
RPE-278: Smoke tests

### PR DESCRIPTION
**Pull request by doncem**
### JIRA link (if applicable) ###

[Implement smoke tests for producer](https://tools.hmcts.net/jira/browse/RPE-278)

### Change description ###

Checking authorisation header is present for all endpoints and it returns 401 for invalid one

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by lgonczar** [comment]
Are there going to be other smoke tests? These ones are valid, but they only check if the service rejects requests in the right cases. What we care about the most is whether it will accept a request and prove it to us by responding to our status requests.

**Comment by doncem** [comment]
During the planning we agreed to just do smoke tests for 400s as it is unclear at the moment how to do more appropriately. As long as smoke tests are present in the pipeline.
I do expect at least one more significant smoke test to check whether actual letter was saved in db/queue - key purpose of the service

File: src/smokeTest/java/uk/gov/hmcts/reform/sendletter/controllers/UnauthorisedSmokeTest.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    @Override
+    public void setup() {
+        super.setup();
+
+        try {
+            createLetterBody = Resources.toString(Resources.getResource("letter.json"), Charsets.UTF_8);
+        } catch (IOException exc) {
+            log.error("Unable to read letter.json from resource path", exc);
+
+            createLetterBody = "";
```
Comment (rkondratowicz) [inline src/smokeTest/java/uk/gov/hmcts/reform/sendletter/controllers/UnauthorisedSmokeTest.java]: just throw that exception and fail the test

**Comment by lgonczar** [comment]
Oh, true.

**Pull request merged**